src/server/pps/server/api_server.go

```diff
@@ -2309,15 +2313,13 @@ func (a *apiServer) deletePipeline(pachClient *client.APIClient, request *pps.De
 		return nil
 	})
 	// Delete cron input repos
-	if pipelineInfo.Input != nil {
-		pps.VisitInput(pipelineInfo.Input, func(input *pps.Input) {
-			if input.Cron != nil {
-				eg.Go(func() error {
-					return pachClient.DeleteRepo(input.Cron.Repo, request.Force)
-				})
-			}
-		})
-	}
+	pps.VisitInput(pipelineInfo.Input, func(input *pps.Input) {
+		if input.Cron != nil {
+			eg.Go(func() error {
+				return pachClient.DeleteRepo(input.Cron.Repo, request.Force)
+			})
+		}
+	})
```

Comment:
This is an unrelated improvement you're able to make now because `VisitInput` correctly handles `nil` right? If so that's a nice bonus, I wonder why we were already defending against `nil` inputs here, I guess at one point we actually allowed them.

Reply:
Yep, that's right